shtdlib.sh

```diff
@@ -910,7 +910,7 @@ function create_secure_tmp {
     dir=${3:-}
     if [ -d "${dir}" ]; then
         if [ "${os_type}" == 'Linux' ]; then
-            secure_tmp_object="$(mktemp ${type_flag} --tmpdir="${dir}" -q )"
+            secure_tmp_object="$(mktemp ${type_flag} -p "${dir}" -q )"
```

Comment:
I see, I didn't know -p created the directory if it didn't exist, I agree the command should fail if it cannot find the directory. I changed this flag in order to pass the Travis CI tests because on busybox the mktemp function didn't have the --tmpdir option so this use case does not work in those docker instances.

Reply:
Ohh sorry, my mistake, I got confused with which command and thought the flag was for mkdir.

If --tmpdir is not supported on busybox then we should probably change to -p as you were doing.